src/engine/sync/syncable.h

```diff
@@ -81,13 +81,25 @@ class Syncable {
 
 class SyncableListener {
   public:
+    virtual ~SyncableListener() {}
+
+    // Used by Syncables to tell EngineSync it wants to be enabled in a
+    // specific mode. If the state change is accepted, EngineSync calls
+    // Syncable::notifySyncModeChanged.
     virtual void requestSyncMode(Syncable* pSyncable, SyncMode mode) = 0;
+
+    // Used by Syncables to tell EngineSync it wants to be enabled in any mode
+    // (master/follower).
     virtual void requestEnableSync(Syncable* pSyncable, bool enabled) = 0;
 
     // A Syncable must never call notifyBpmChanged in response to a setMasterBpm()
     // call.
     virtual void notifyBpmChanged(Syncable* pSyncable, double bpm,
```

Comment:
Why not a separate signal notifyFileBpmChanged, like notifyInstantaneousBpmChanged? We should either use a signal signal with an enumeration parameter or separate signals, but not a mixture of both style.

Separate signals seem to make sense, because we also have separate slots for the different use cases.

Reply:
OK, I have refactored the call a bit. I hope I have got your point.